### PR TITLE
[utils] append the original error to the error string

### PR DESCRIPTION
**Pull request by mayorova**

### CODE REVIEW

File: gateway/src/apicast/threescale_utils.lua [hunk 1]
```diff
@@ -152,7 +152,7 @@ function _M.connect_redis(options)
 
   local ok, err = red:connect(_M.resolve(host, port))
   if not ok then
-    return nil, _M.error("failed to connect to redis on " .. host .. ":" .. port .. ":", err)
+    return nil, _M.error("failed to connect to redis on " .. host .. ":" .. port .. ": " .. err)
```
[the remark this answers is not in the document]
Comment (mikz) [reply gateway/src/apicast/threescale_utils.lua]: We should avoid string concatenation at all costs. This is in error handling, so not so frequent, but still we should follow the same pattern everywhere.

As @davidor says `_M.error` will pass all the arguments to `ngx.say` that is going to print them to the output.
https://github.com/3scale/apicast/blob/1415f0abdc35aca9666ff4db76cdcb91b17f7cce/gateway/src/apicast/threescale_utils.lua#L191-L199

Proper fix would be to change all string concatenation to this pattern instead.

```lua
_M.error("failed to connect to redis on ", host, ":", port, ": ", err)
```

[the remark this answers is not in the document]
Comment (mayorova) [reply gateway/src/apicast/threescale_utils.lua]: Well, this is problematic, as if you do:
```
local redis, err = ts.connect_redis{ url = redis_url }
ngx.log(ngx.ERR, err)
```
the `err` variable will get only the first piece.

And I'd say this is the most common way to capture errors, so I don't know if there are any alternatives...

[the remark this answers is not in the document]
Comment (mikz) [reply gateway/src/apicast/threescale_utils.lua]: then `_M.error` can `return nil, table.concat(' ', select(1, ...))`

[the remark this answers is not in the document]
Comment (mayorova) [reply gateway/src/apicast/threescale_utils.lua]: how about `table.concat({...})`? (committed)

**Comment by mikz** [comment]
@mayorova pushed a test and a tiny cleanup

**Comment by mayorova** [comment]
Great, thank you @mikz :)

**Pull request merged**